apps/passport-server/test/generic-issuance/credentialSubservice.spec.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  EdDSAPublicKey,
+  getEdDSAPublicKey,
+  isEqualEdDSAPublicKey,
+  newEdDSAPrivateKey
+} from "@pcd/eddsa-pcd";
+import { EmailPCDPackage } from "@pcd/email-pcd";
+import {
+  PODBOX_CREDENTIAL_REQUEST,
+  VerificationError,
+  ZUPASS_CREDENTIAL_REQUEST,
+  createCredentialPayload
+} from "@pcd/passport-interface";
+import { ONE_DAY_MS } from "@pcd/util";
+import { Identity } from "@semaphore-protocol/identity";
+import { expect } from "chai";
+import "mocha";
+import { step } from "mocha-steps";
+import MockDate from "mockdate";
+import { CredentialSubservice } from "../../src/services/generic-issuance/subservices/CredentialSubservice";
+import { overrideEnvironment, testingEnv } from "../util/env";
+import { startTestingApp } from "../util/startTestingApplication";
+import { expectToExist } from "../util/util";
+import { makeCredential, proveEmailPCD, signCredentialPayload } from "./util";
+
+/**
+ * Tests for {@link CredentialSubservice}
+ */
+describe("generic issuance - credential subservice", function () {
+  let credentialSubservice: CredentialSubservice;
+  let ZUPASS_EDDSA_PRIVATE_KEY: string;
+  let zupassPublicKey: EdDSAPublicKey;
+
+  /**
+   * Sets up a Zupass/Generic issuance backend.
+   */
+  this.beforeAll(async () => {
+    await overrideEnvironment({
+      ...testingEnv,
+      GENERIC_ISSUANCE_ZUPASS_PUBLIC_KEY: JSON.stringify(zupassPublicKey),
+      STYTCH_BYPASS: "true",
+      NODE_ENV: "test"
+    });
+
+    ZUPASS_EDDSA_PRIVATE_KEY = process.env.SERVER_EDDSA_PRIVATE_KEY as string;
+    zupassPublicKey = await getEdDSAPublicKey(ZUPASS_EDDSA_PRIVATE_KEY);
+
+    await startTestingApp({});
+    credentialSubservice = new CredentialSubservice(zupassPublicKey);
+  });
+
+  this.afterAll(() => {
+    MockDate.reset();
+  });
+
+  step("credential subservice can verify credentials", async () => {
+    const identity = new Identity();
+    const emailAddress = "test@example.com";
+    {
+      const credential = await makeCredential(
```

Comment:
suggest naming `makeCredential` `makeCredentialForTest` to signal that it's a utility method not intended to be used by actual application code